### ISSUE TITLE
TypeError when IDP response does not include signing certificate

### ISSUE THREAD

**Issue by makizm** [tngan/samlify#231]
libsaml.ts throws type error on signature verification.

`TypeError: Cannot read property 'firstChild' of undefinedOD`

It appears 'verifySignature' is expecting IDP to include signing certificate in the response. When the signature is not included `select ()` returns empty array so the next line throws a TypeError.

https://github.com/tngan/samlify/blob/4736dd5d2451bc08089bfc5f7fef4502daeb9885/src/libsaml.ts#L376-L377

When IDP is configured to include signing certificate in the response (see below IDP response) 'verifySignature' is happy all the sudden.

**IDP response**
```xml
<ds:Signature xmlns:ds="http://www.w3.org/2000/09/xmldsig#">
   <ds:KeyInfo>
     <ds:X509Data>
       <ds:X509Certificate>
         MII..
       </ds:X509Certificate>
     </ds:X509Data>
   </ds:KeyInfo>
</ds:Signature>
<ds:SignatureValue>
  signaturevalue
</ds:SignatureValue>
```

IDP signing certificate is already specified in the metadata xml file so including signature in the IDP response should not be required.

**Comment by eugene1g** [comment]
Related #223 (but I never got to doing the PR)

**Comment by makizm** [comment]
Here is a PR to fix type error https://github.com/tngan/samlify/pull/232

For anyone that comes across this just make sure to configure your IDP to include signing ssl cert with responses and all will work as expected.

Hope to see verification done based IdentityProvider metadata!

**Comment by tngan** [comment]
The fix was released with v2.4.2.
